packages/core-utils/src/watcher.ts

```diff
@@ -61,10 +61,11 @@ export class Watcher {
     for (const log of receipt.logs) {
       if (
         log.address === layer.messengerAddress &&
-        log.topics[0] === ethers.utils.id('SentMessage(bytes)')
+        log.topics[0] ===
+          ethers.utils.id('SentMessage(address,address,bytes,uint256,uint256)')
       ) {
-        const [message] = ethers.utils.defaultAbiCoder.decode(
-          ['bytes'],
+        const [, , message, ,] = ethers.utils.defaultAbiCoder.decode(
+          ['address', 'address', 'bytes', 'uint256', 'uint256'],
           log.data
         )
         msgHashes.push(ethers.utils.solidityKeccak256(['bytes'], [message]))
```

Comment:
Wrong value being used here. Hash needs to be computed from the event parameters as you did with encodedMessageHash below. Probably contributing to at least some of the test failures.

Reply:
There were issues associated with the fact that indexing an event parameter causes the indexed property to move from `logs.data` to `logs.topics` so had to adjust the watcher for this. But now there remain 7 failing tests to do with L2 to L1 transactions that I cannot get to the bottom of. Can you have a look pls @smartcontracts

Reply:
Am reviewing this now. Sorry for the delay here.